docs/DeveloperGuide.md

```diff
@@ -2,36 +2,524 @@
+
+As for registering new users, newly inputted username and password will be saved to the users attribute and this
+pair of username and password is then appended to the txt file containing current users. The updated user list will be
+loaded into the users attribute when the application is booted up again.
+
+The login system class uses the below methods to achieve its functionality
+
+- `authenticateUserChoice()` -Decides whether the user chooses to register or login
+- `newUserCreator()` -Creates a new user for future login
+- `loginExistingUser()` -Login existing user
+- `loadExistingUsers()` -Load existing users into hashtable for reference
+
+Given below is an example of how the login system class works.
+
+When the user first launches the programme, the Stocker object will be instantiated. The object will
+invoke its own `run()` method which will call its own `start()` method. The Stocker object then instantiates a
+new UI object which displays the login message by invoking `showLoginMessage()` method. At this point, Stocker object
+will also instantiate a new login system object.
+
+The login system object will invoke its own `run()` method to begin the login process. This method begins by loading
+existing users into the users attribute of the login system class by `loadExistingusers()` method. it then invokes
+`authenticateUserChoice()` to receive an input from the user to whether register or login a user. Based on the input of
+the user, either `newUserCreator()` is launched or `loginExistingUser()` methods will be called to register or login a
+user.
+
+The following sequence diagram shows how the login system class works when the program is launched.
+
+<img src="UML Diagrams/StockerToLoginSystem.png" width="280">
+
+---
+
+### Parser Component
+
+---
+
+The Parser component is responsible for interpreting user input and converting it into executable commands. It plays a
+critical role in bridging the user interface (UI) and the command execution components of the application.
+
+#### Design considerations
+
+- **User Input Parsing:** The Parser must effectively break down user input into its constituent parts, such as the command
+keyword and any associated arguments.
+
+- **Command Recognition:** The Parser must recognize the specific command the user intends to execute. This involves matching
+the command keyword to a predefined set of commands.
+
+- **Arguments Extraction:** For commands that require additional information, the Parser should correctly extract and format
+arguments, ensuring they are ready for command execution.
+
+- **Error Handling:** In cases where the input does not match any recognized command or has formatting errors, the Parser
+should generate appropriate error messages.
+
+
+#### Implementation and rationale
+
+The Parser class is designed to handle these considerations through a well-structured parsing process. Here's how it
+works:
+
```

Comment:
Would be clearer to reader if you bold "Splitting User Input", "Command Recognition" etc.